### PR TITLE
feat: add recovery words vault 

### PR DESCRIPTION
**Pull request by imGok**
https://github.com/linagora/twake-on-matrix/assets/6462404/2a5d7cff-84ed-446c-aade-563e2c5c0538

### CODE REVIEW

File: lib/domain/model/tom_server_information.g.dart
Comment (hoangdat) [inline lib/domain/model/tom_server_information.g.dart:1]: please remove this file

File: linux/flutter/generated_plugins.cmake
Comment (hoangdat) [inline linux/flutter/generated_plugins.cmake:1]: remove this also

File: linux/flutter/generated_plugin_registrant.cc
Comment (hoangdat) [inline linux/flutter/generated_plugin_registrant.cc:1]: remove this also

File: lib/data/repository/recovery_words_repository.dart
Comment (hoangdat) [inline lib/data/repository/recovery_words_repository.dart:1]: this file should belong to `domain`

File: lib/pages/login/login.dart [hunk 1]
```diff
@@ -72,9 +78,7 @@ class LoginController extends State<Login> {
             identifier: identifier,
             // To stay compatible with older server versions
             // ignore: deprecated_member_use
-            user: identifier.type == AuthenticationIdentifierTypes.userId
-                ? username
-                : null,
+            user: identifier.type == AuthenticationIdentifierTypes.userId ? username : null,
```
Comment (hoangdat) [inline lib/pages/login/login.dart:81]: please dont touch to the format

File: lib/pages/login/login.dart [hunk 2]
```diff
@@ -108,11 +112,24 @@ class LoginController extends State<Login> {
       Matrix.of(context).getLoginClient().homeserver = newDomain;
       DiscoveryInformation? wellKnownInformation;
       try {
-        wellKnownInformation =
-            await Matrix.of(context).getLoginClient().getWellknown();
+        wellKnownInformation = await Matrix.of(context).getLoginClient().getWellknown();
```
Comment (hoangdat) [inline lib/pages/login/login.dart:115]: idem

File: lib/pages/login/login.dart [hunk 3]
```diff
@@ -128,8 +145,7 @@ class LoginController extends State<Login> {
           final dialogResult = await showOkCancelAlertDialog(
             context: context,
             useRootNavigator: false,
-            message:
-                L10n.of(context)!.noMatrixServer(newDomain, oldHomeserver!),
+            message: L10n.of(context)!.noMatrixServer(newDomain, oldHomeserver!),
```
Comment (hoangdat) [inline lib/pages/login/login.dart:148]: idem

File: lib/pages/login/login.dart [hunk 4]
```diff
@@ -166,8 +215,7 @@ class LoginController extends State<Login> {
       fullyCapitalizedForMaterial: false,
       textFields: [
         DialogTextField(
-          initialText:
-              usernameController.text.isEmail ? usernameController.text : '',
+          initialText: usernameController.text.isEmail ? usernameController.text : '',
```
Comment (hoangdat) [inline lib/pages/login/login.dart:218]: idem

File: lib/pages/login/login.dart [hunk 5]
```diff
@@ -177,12 +225,11 @@ class LoginController extends State<Login> {
...
-          Matrix.of(context).getLoginClient().requestTokenToResetPasswordEmail(
-                clientSecret,
-                input.single,
-                sendAttempt++,
-              ),
+      future: () => Matrix.of(context).getLoginClient().requestTokenToResetPasswordEmail(
+            clientSecret,
+            input.single,
+            sendAttempt++,
+          ),
```
Comment (hoangdat) [inline lib/pages/login/login.dart:232]: idem

File: lib/pages/login/login.dart [hunk 6]
```diff
@@ -108,11 +112,24 @@ class LoginController extends State<Login> {
...
+        }
+        if (wellKnownInformation.additionalProperties.containsKey('t.server')) {
+          final toMServer = wellKnownInformation.additionalProperties['t.server'];
+          _setUpToMServer(
+            ToMServerInformation(
+              baseUrl: toMServer!['base_url'],
+              serverName: toMServer['server_name'],
+            ),
+          );
+        }
```
Comment (hoangdat) [inline lib/pages/login/login.dart:132]: Thanks for this, but IMO, you should focus to your logic first

File: lib/data/datasource_impl/recovery_words_data_source_impl.dart [hunk 7]
```diff
@@ -0,0 +1,21 @@
+
+import 'package:fluffychat/data/datasource/recovery_words_data_source.dart';
+import 'package:fluffychat/data/network/dio_client.dart';
+import 'package:fluffychat/domain/model/message_response.dart';
+import 'package:fluffychat/domain/model/recovery_words_json.dart';
+
+class RecoveryWordsDataSourceImpl implements RecoveryWordsDataSource {
+  final DioClient dioClient;
```
Comment (hoangdat) [inline lib/data/datasource_impl/recovery_words_data_source_impl.dart:8]:
```suggestion
  final DioClient dioClient = getIt.get(instanceName: tomServerDioName);
```

**Comment by hoangdat** [comment]
No demo, so hard for us to understand

File: lib/data/repository/recovery_words_repository_impl.dart [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import 'package:fluffychat/data/datasource/recovery_words_data_source.dart';
+import 'package:fluffychat/data/model/recovery_words_json.dart';
+import 'package:fluffychat/domain/repository/recovery_words_repository.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/domain/model/recovery_words/recovery_words.dart';
+
+class RecoveryWordsRepositoryImpl implements RecoveryWordsRepository {
+  final RecoveryWordsDataSource recoveryWordsDataSource = getIt.get<RecoveryWordsDataSource>();
```
Comment (sherlockvn) [inline lib/data/repository/recovery_words_repository_impl.dart:8]: use `final recoveryWordsDataSource = getIt.get<RecoveryWordsDataSource>();` for shorter version

File: lib/data/network/recovery_words/recovery_words_api.dart [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import 'package:fluffychat/data/model/recovery_words_json.dart';
+import 'package:fluffychat/data/network/dio_client.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/di/global/network_di.dart';
+
+class RecoveryWordsAPI {
+  final DioClient _client = getIt.get<DioClient>(instanceName: NetworkDI.tomDioClientName);
```
Comment (sherlockvn) [inline lib/data/network/recovery_words/recovery_words_api.dart:7]: use `final _client = getIt.get<DioClient>(instanceName: NetworkDI.tomDioClientName);`  for shorter version

**Pull request closed without merging**